File: database_ui/src/types/load.ts
export interface Load {
  load_id: string;
  origin: string;
  destination: string;
  pickup_datetime: string;
  delivery_datetime: string;
  equipment_type: string;
  loadboard_rate: number;
  notes: string | null;
  weight: number | null;
  commodity_type: string | null;
  num_of_pieces: number | null;
  miles: number | null;
  dimensions: string | null;
  status: string;
}

export interface LoadsResponse {
  loads: Load[];
  total: number;
  limit: number;
  offset: number;
}

export interface LoadCreate {
  load_id: string;
  origin: string;
  destination: string;
  pickup_datetime: string;
  delivery_datetime: string;
  equipment_type: string;
  loadboard_rate: number;
  notes?: string;
  weight?: number;
  commodity_type?: string;
  num_of_pieces?: number;
  miles?: number;
  dimensions?: string;
}

export const EQUIPMENT_TYPES = ["Dry Van", "Reefer", "Flatbed", "Step Deck"] as const;

File: api/app/schemas/load.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class LoadOut(BaseModel):
    load_id: str
    origin: str
    destination: str
    pickup_datetime: datetime
    delivery_datetime: datetime
    equipment_type: str
    loadboard_rate: Decimal
    notes: str | None
    weight: int | None
    commodity_type: str | None
    num_of_pieces: int | None
    miles: int | None
    dimensions: str | None
    status: str

    model_config = {"from_attributes": True}


class LoadsResponse(BaseModel):
    loads: list[LoadOut]
    total: int
    limit: int
    offset: int
